internal/resolution/variablesources/bundle.go

```diff
@@ -2,62 +2,36 @@ package variablesources
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"sort"
 
+	"k8s.io/apimachinery/pkg/util/sets"
+
 	"github.com/operator-framework/deppy/pkg/deppy"
 	"github.com/operator-framework/deppy/pkg/deppy/input"
-	"k8s.io/apimachinery/pkg/util/sets"
 
 	"github.com/operator-framework/operator-controller/internal/catalogmetadata"
 	catalogfilter "github.com/operator-framework/operator-controller/internal/catalogmetadata/filter"
 	catalogsort "github.com/operator-framework/operator-controller/internal/catalogmetadata/sort"
 	olmvariables "github.com/operator-framework/operator-controller/internal/resolution/variables"
 )
 
-var _ input.VariableSource = &BundlesAndDepsVariableSource{}
-
-type BundlesAndDepsVariableSource struct {
-	catalogClient   BundleProvider
-	variableSources []input.VariableSource
-}
-
-func NewBundlesAndDepsVariableSource(catalogClient BundleProvider, inputVariableSources ...input.VariableSource) *BundlesAndDepsVariableSource {
-	return &BundlesAndDepsVariableSource{
-		catalogClient:   catalogClient,
-		variableSources: inputVariableSources,
+func MakeBundleVariables(
+	allBundles []*catalogmetadata.Bundle,
+	requiredPackages []*olmvariables.RequiredPackageVariable,
+	installedPackages []*olmvariables.InstalledPackageVariable,
+) ([]*olmvariables.BundleVariable, error) {
+	bundleQueue := make([]*catalogmetadata.Bundle, 0, len(requiredPackages)+len(installedPackages))
```

Comment:
`len(requiredPackages)+len(installedPackages)` is an inexact approximation of the size you need, since each of those has a 1:N relationship to the underlying bundles? Personal preference, but `var bundleQueue []*catalogmetadata.Bundle` is fine unless benchmarks prove otherwise. Pointers are small.

Reply:
That was initial version and is my preference too. We had a discussion about this here: https://github.com/operator-framework/operator-controller/pull/498#discussion_r1380044187

I'm happy to revert this back to `var bundleQueue []*catalogmetadata.Bundle`.

@joelanford any objections?

Reply:
I think @stevekuznetsov and I have pretty much the exact same perspective, and we just fell on either side of the fence in terms of what to do.

In my mind, we _know_ `len(requiredPackages)+len(installedPackages)` is a strictly better approximation than `0`, so that's why I fell on that side.

I also prefaced my comment with "this may not matter, benchmarks would help", so I don't have a strong opinion.

Maybe the outcome of this should be a follow-on issue that says "let's write some benchmarks and have a way to measure how we're doing".

Reply:
My perspective is that a couple things have to be true before it's worth doing something like that:

a) the code in question is performance sensitive
b) we have a way to measure it
c) the naive performance is insufficient

I know loads of performance sensitive code where either _N_ is small and the few copies don't matter or the majority of time is taken on other work, so even a 99% speedup in alloc makes negligible/unmeasureably small impact on the overall runtime.

The reason _not_ to do this unless the prerequisites are met is that it's easy to do subtly incorrect things with this - especially when performance sensitive code uses pre-allocated length with `[i]` member access instead of `append`, so not asking readers to have the mental overhead unless we know we need to seems good.

Feel free to keep the code here as it is, as I don't feel super strongly on this, but overall IMHO we benefit from KISS.